src/rgw/rgw_auth.h

```diff
@@ -430,28 +435,32 @@ class LocalApplier : public IdentityApplier {
 
 protected:
   const RGWUserInfo user_info;
-  const std::string subuser;
-
-  uint32_t get_perm_mask(const std::string& subuser_name,
-                         const RGWUserInfo &uinfo) const;
+  std::string subuser;
```

Comment:
De facto we're caching the information here. Is this because of performance reasons? Do we expect to handle users with really huge number of subusers inside?